common/beerocks/bwl/dwpal/mon_wlan_hal_dwpal.cpp

```diff
@@ -584,6 +725,105 @@ bool mon_wlan_hal_dwpal::process_dwpal_event(char *buffer, int bufLen, const std
 
 bool mon_wlan_hal_dwpal::process_dwpal_nl_event(struct nl_msg *msg)
 {
+    struct nlmsghdr *nlh     = nlmsg_hdr(msg);
+    struct genlmsghdr *gnlh  = (genlmsghdr *)nlmsg_data(nlh);
+    char ifname[IF_NAMESIZE] = "\0";
+    struct nlattr *tb[NL80211_ATTR_MAX + 1];
+    nla_parse(tb, NL80211_ATTR_MAX, genlmsg_attrdata(gnlh, 0), genlmsg_attrlen(gnlh, 0), NULL);
+
+    if (tb[NL80211_ATTR_IFINDEX] != NULL) {
+        if_indextoname(nla_get_u32(tb[NL80211_ATTR_IFINDEX]), ifname);
+    }
+
+    auto event = dwpal_nl_to_bwl_event(gnlh->cmd, m_waiting_for_results_ready);
+
+    switch (event) {
+    case Event::Channel_Scan_Triggered: {
+        LOG(DEBUG) << "DWPAL NL event channel scan triggered";
+        if (m_radio_info.iface_name.compare(ifname) != 0) {
+            // ifname doesn't match current interface
+            // meaning the event was recevied for a diffrent channel
+            LOG(DEBUG) << "ignoring event for other interfaces";
+            return true;
+        }
+
+        m_waiting_for_results_ready = true;
```

Comment:
This requires some explanation. Why receiving "Channel_Scan_Triggered" event sets this flag to "true", and where it's used later on.

Reply:
done